services/core-web/common/models/documents/document.ts

```diff
@@ -150,25 +200,13 @@ export class MajorMineApplicationDocument extends MineDocument {
     } = jsonObject;
 
     return (
-      project_summary_document_xref?.project_summary_document_type_code
-      ?? project_decision_package_document_xref?.project_decision_package_document_type_code
-      ?? information_requirements_table_document_xref?.information_requirements_table_document_type_code
-      ?? major_mine_application_document_xref?.major_mine_application_document_type_code
+      project_summary_document_xref?.project_summary_document_type_code ??
+      project_decision_package_document_xref?.project_decision_package_document_type_code ??
+      information_requirements_table_document_xref?.information_requirements_table_document_type_code ??
+      major_mine_application_document_xref?.major_mine_application_document_type_code
     );
   }
 
-  protected makeChild(params: any, constructorArgs: any) {
```

Comment:
I think there will be some difficulty in making versions a different type from the latest version. The file permissions are based on the type, and it keeps the properties consistent for document tables. If it works, that'd be a nice bit of clean-up, but I would be surprised if the document tables still have the right column data and allowed actions based on user roles.

Reply:
The data in the columns is still displaying correctly for the versions actually 😄 

They have a handful of common fields between MineDocuments and MineDocumentVersions but passing the version info into a MineDocument was mostly just ending up with a bunch of undefined fields.  Since versions should always conform to the same structure, I think it makes sense to simplify them like this.

I did add the allowed actions for versions, since they should always just be download and or view.

Reply:
Ahh, that's right, we took out the File Location column that displays the major_mine_application_type_code, so it wouldn't produce an error with the columns we're using. If we're not planning on displaying versions with columns that are specific to the type then this shouldn't be a problem.

Might still suggest passing the view/download permission from the parent, so that we'd be able to restrict that based on permission (should be just as easy). I thought that the child data still wanted a unique key though- are you not getting a "Each record in table should have a unique key prop" message? No issues sorting? <-- that's when I ran into weird bugs

Reply:
Hmmm, let me check the key issue.

Sure, we can pass the view/download permission from the parent.  Do we ever restrict this though?  I thought we only restricted the other functions.